src/CoreBundle/private/library/classes/TCMSFields/TCMSFieldExtendedLookup.class.php

```diff
@@ -152,6 +150,58 @@ protected function _GetOpenWindowJS(&$oPopupTableConf)
         return $js;
     }
 
+    private function getTargetListRestriction(): array
+    {
+        $restrictionExpression = $this->oDefinition->GetFieldtypeConfigKey('restriction');
+        if (null === $restrictionExpression) {
+            return [];
+        }
+        if (false === \strpos($restrictionExpression, '=')) {
+            return [];
+        }
+        $equalPosition = \mb_strpos($restrictionExpression, '=');
+        $restrictionField = \trim(\mb_substr($restrictionExpression, 0, $equalPosition));
+        $restrictionValue = \trim(\mb_substr($restrictionExpression, $equalPosition + 1));
+
+        if (\mb_strpos($restrictionValue, '[{') === 0 && '}]' === \mb_substr($restrictionValue, -2)) {
+            $restrictionValueReplaced = $this->getFieldValueFromFieldToken($restrictionValue);
+            if (null === $restrictionValueReplaced) {
+                $restrictionValueReplaced = \sprintf('field token %s not found in record.', $restrictionValue);
+            }
+            $restrictionValue = $restrictionValueReplaced;
+        }
+
+        if ('' === $restrictionField || '' === $restrictionValue) {
+            return [];
+        }
+
+        return [
+            'sRestrictionField' => $restrictionField,
+            'sRestriction' => $restrictionValue,
+        ];
+    }
+
+    /**
+     * Returns field value in current record if token matches a known field. Otherwise, null will be returned.
+     * @param string $fieldToken - token in the form [{fieldName}]
+     * @return string|null
+     */
+    private function getFieldValueFromFieldToken(string $fieldToken): ?string
+    {
+        if (false === \property_exists($this, 'oTableRow')) {
+            return null;
+        }
+        if (false === \is_object($this->oTableRow)) {
+            return null;
+        }
+        if (false === \is_array($this->oTableRow->sqlData)) {
+            return null;
+        }
+        $fieldName = \trim(\mb_substr($fieldToken, 2, -2));
+
+        return $this->oTableRow->sqlData[$fieldName] ?? null;
```

Comment:
Wouldn't these two lines be sufficient here (esp. ?? )?

Reply:
You are right. I thought that the null coalescing operator would not work on nested properties and that I would need the nullsafe operator. Turns out I would need that only if there was a method call in the chain.